lib/livebook/hubs.ex

```diff
@@ -98,13 +98,48 @@ defmodule Livebook.Hubs do
     with {:ok, hub} <- fetch_hub(id) do
       true = Provider.type(hub) != "personal"
       :ok = Broadcasts.hub_changed(hub.id)
+      :ok = maybe_unset_default_hub(hub.id)
       :ok = Storage.delete(@namespace, id)
       :ok = disconnect_hub(hub)
     end
 
     :ok
   end
 
+  @spec set_default_hub(String.t()) :: :ok
+  def set_default_hub(id) do
+    with {:ok, hub} <- fetch_hub(id) do
+      :ok = Broadcasts.default_hub_changed(hub.id)
+      :ok = Storage.insert(:default_hub, "default_hub", [{:default_hub, hub.id}])
```

Comment:
First insert, then broadcast, so if anyone reads after the broadcast they get the up to date data.

```suggestion
      :ok = Storage.insert(:default_hub, "default_hub", [{:default_hub, hub.id}])
      :ok = Broadcasts.default_hub_changed(hub.id)
```